### PR TITLE
feat: contributor add share feature

### PR DESCRIPTION
**Pull request by LiteSun**

### CODE REVIEW

File: views/Contributor/Contributor.tsx [hunk 1]
```diff
@@ -22,9 +29,50 @@ const getParameterByName = (name, url = '') => {
...
+        event="click"
+        mainButtonStyles={{ background: "#1DB954" }}
+        alwaysShowTitle={true}
+        icon={<ShareIcon />}
+      >
+        <Action
+          text="Share on Twitter"
+          style={{ backgroundColor: "rgb(29, 161, 242)" }}
+          onClick={() => {
+            window.location.href = `https://twitter.com/share?text=Amazing tools to view your repo contributor over time!&url=https://www.apiseven.com/zh/contributor-graph?repo=${legend.join(
```
Comment (juzhiyuan) [inline views/Contributor/Contributor.tsx:52]: Why use text in English but with `Chinese` site?

File: views/Contributor/Contributor.tsx [hunk 2]
```diff
@@ -22,9 +29,50 @@ const getParameterByName = (name, url = '') => {
 
 const Contributor: NextPage<Props, any> = ({ t, url }) => {
   const repo = getParameterByName('repo', url);
+  const [legend, setLegend] = useState([]);
+
+  useEffect(() => {
+    window.addEventListener('message', function (event) {
+      setLegend(event.data);
+    })
+  });
```
Comment (juzhiyuan) [inline views/Contributor/Contributor.tsx:38]:
```suggestion
  }, []);
```

File: views/Contributor/Contributor.tsx [hunk 3]
```diff
@@ -22,9 +29,50 @@ const getParameterByName = (name, url = '') => {
...
+            )}`;
+          }}
+        >
+          <TwitterIcon />
+        </Action>
+        <Action
+          text="Copy share link"
+          style={{ backgroundColor: "#1769FF" }}
+          onClick={() => {
+            const text = `${window.location.protocol +
```
Comment (juzhiyuan) [inline views/Contributor/Contributor.tsx:63]: Why not use `location.href`?

Comment (LiteSun) [reply views/Contributor/Contributor.tsx:63]: `location.href` contains query info. we need manual splicing link

**Review by juzhiyuan — APPROVED** [review]

**Comment by juzhiyuan** [comment]
image

**Comment by juzhiyuan** [comment]
When clicking Share on Twitter, the current page will get refreshed. @LiteSun

**Comment by LiteSun** [comment]
> When clicking Share on Twitter, the current page will get refreshed. @LiteSun

we need to use `window.open` ?

**Comment by juzhiyuan** [comment]
ya

**Pull request merged**